suite-native/module-send/src/components/ReviewOutputItemList.tsx

```diff
@@ -57,26 +62,36 @@ export const ReviewOutputItemList = ({ accountKey, tokenContract }: ReviewOutput
 
     if (!account) return <ErrorMessage errorMessage="Account not found." />;
 
+    const shouldShowAllOutputs = buttonRequestCount > 0 || isTransactionAlreadySigned;
+    const shouldShowFooter =
+        !isTransactionAlreadySigned && buttonRequestCount > 0 && (reviewOutputs?.length ?? 0) > 0;
+
     return (
         <>
             {reviewOutputs && (
                 <VStack spacing={LIST_VERTICAL_SPACING}>
-                    {reviewOutputs?.map((output, index) => (
-                        <ReviewOutputItem
-                            key={output.value}
-                            reviewOutput={output}
-                            onLayout={event => handleReadListItemHeight(event, index)}
+                    {reviewOutputs
+                        ?.filter((_, index) => shouldShowAllOutputs || index === 0)
+                        .map((output, index) => (
+                            <ReviewOutputItem
+                                key={`${output.type}-${output.value}`}
+                                reviewOutput={output}
+                                onLayout={event => handleReadListItemHeight(event, index)}
+                            />
+                        ))}
+                    {shouldShowAllOutputs && (
+                        <ReviewOutputSummaryItem
+                            accountKey={accountKey}
+                            symbol={account.symbol}
+                            tokenContract={tokenContract}
+                            onLayout={event =>
+                                handleReadListItemHeight(event, reviewOutputs.length)
+                            }
                         />
-                    ))}
-                    <ReviewOutputSummaryItem
-                        accountKey={accountKey}
-                        symbol={account.symbol}
-                        tokenContract={tokenContract}
-                        onLayout={event => handleReadListItemHeight(event, reviewOutputs.length)}
-                    />
+                    )}
```

Comment:
I am wondering... Would it make sense to not render anything until `shouldShowAllOutputs` is true? filtering the list and conditionally displaying the last element seems quite hacky to me. 

e.g.

```suggestion
                    
                    {shouldRenderItems ? (
                        reviewOutputs
                        .map((output, index) => (
                            <ReviewOutputItem
                                key={`${output.type}-${output.value}`}
                                reviewOutput={output}
                                onLayout={event => handleReadListItemHeight(event, index)}
                            />
                        ))
                        <ReviewOutputSummaryItem
                            accountKey={accountKey}
                            symbol={account.symbol}
                            tokenContract={tokenContract}
                            onLayout={event =>
                                handleReadListItemHeight(event, reviewOutputs.length)
                            }
                        />
                    ) : <Spinner/>
                    }
                    
```

WDYT?